internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestBatch.java

```diff
@@ -1336,4 +1340,60 @@ public void testKafkaInputBatchJobSucceedsWhenSourceTopicIsEmpty() throws Except
       Assert.assertEquals(version, 2);
     });
   }
+
+  @Test(timeOut = TEST_TIMEOUT, dataProvider = "True-and-False", dataProviderClass = DataProviderUtils.class)
+  public void testBatchJobSnapshots(Boolean isKakfaPush) throws Exception {
+
+    VPJValidator validator = (avroClient, vsonClient, metricsRepository) -> {
+      for (int i = 1; i <= 100; i++) {
+        Assert.assertEquals(avroClient.get(Integer.toString(i)).get().toString(), "test_name_" + i);
+      }
+    };
+
+    String storeName = testBatchStore(
+        inputDir -> new KeyAndValueSchemas(writeSimpleAvroFileWithStringToStringSchema(inputDir)),
+        properties -> {},
+        validator);
+
+    verifySnapshotFolders();
```

Comment:
Do you think we can clean up directories before we conduct 2nd push so we can ensure the 2nd push also creates the snapshots?